src/workflow_config.py

```
@@ -0,0 +1,188 @@
+        config[keys[-1]] = value
+
+    def __str__(self) -> str:
+        """String representation of configuration."""
+        return json.dumps(self._config, indent=2)
+
+
+# Global configuration instance
+_workflow_config = None
+
+
+def get_workflow_config(config_file: Path = None) -> WorkflowConfig:
```

Ruff (0.14.3)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)